code/modules/atmospherics/machinery/atmosmachinery.dm

```diff
@@ -228,7 +233,8 @@
 /obj/machinery/atmospherics/proc/isConnectable(obj/machinery/atmospherics/target, given_layer)
 	if(isnull(given_layer))
 		given_layer = piping_layer
-	if((target.piping_layer == given_layer) || (target.pipe_flags & PIPING_ALL_LAYER))
+	if(	((target.piping_layer == given_layer) || (target.pipe_flags & PIPING_ALL_LAYER)) && target.loc != loc && \
+		(target.pipe_color == pipe_color || target.pipe_flags & PIPING_ALL_COLORS || pipe_flags & PIPING_ALL_COLORS || target.pipe_color == rgb(254,254,254) || pipe_color == rgb(254,254,254)))
```

Comment:
```suggestion
	if(
		((target.piping_layer == given_layer) || (target.pipe_flags & PIPING_ALL_LAYER)) && target.loc != loc \
		&& (target.pipe_color == pipe_color || target.pipe_flags & PIPING_ALL_COLORS || pipe_flags & PIPING_ALL_COLORS || target.pipe_color == rgb(254,254,254) || pipe_color == rgb(254,254,254))
	)
```

If you're gonna do multi-lining, then it's gotta be correct, no random indentations.

The `&&` on its own line makes it easier to know what the next operation is without going to the end of the previous line.